src/lib/communication/DY.Lib.Communication.Core.Invariants.fst

```diff
@@ -0,0 +1,188 @@
+module DY.Lib.Communication.Core.Invariants
+
+open Comparse
+open DY.Core
+open DY.Lib.Crypto.PkEncryption.Split
+open DY.Lib.Crypto.Signature.Split
+open DY.Lib.Crypto.AEAD.Split
+open DY.Lib.Event.Typed
+open DY.Lib.State.PrivateKeys
+open DY.Lib.State.Tagged
+
+open DY.Lib.Communication.Core
+
+#set-options "--fuel 0 --ifuel 0 --z3cliopt 'smt.qi.eager_threshold=100'"
+
+(*** PkEnc Predicates ***)
+
+val pkenc_crypto_predicates_communication_layer: {|cusages:crypto_usages|} -> pkenc_crypto_predicate
+let pkenc_crypto_predicates_communication_layer #cusages = {
+  pred = (fun tr sk_usage pk msg ->
+    (exists sender receiver.
+      sk_usage == long_term_key_type_to_usage (LongTermPkEncKey comm_layer_pkenc_tag)  receiver /\
+      (get_label tr msg) `can_flow tr` (join (principal_label sender) (principal_label receiver)) /\
```

Comment:
This can probably use `communication_label sender receiver` now that that's defined.